### PR TITLE
Create Multithreading in Java

### PR DESCRIPTION
**Pull request by XavierQuan**
I have created a documentation on the topic of multithreading in Java, which includes core concepts and some pratical instructions.

### CODE REVIEW

**Review by ChenJiang03 — COMMENTED** [review]
This is a very detailed explanation of multithreading. However, from my current understanding, data sharing between threads is a key point in correctly utilizing multithreading to its max potential. As you mentioned above, thread share file, heap, and code. Can you dive deeper into how threads share data between each other?

**Comment by XavierQuan** [comment]
> This is a very detailed explanation of multithreading. However, from my current understanding, data sharing between threads is a key point in correctly utilizing multithreading to its max potential. As you mentioned above, thread share file, heap, and code. Can you dive deeper into how threads share data between each other?

Thank you for your advice and I agree with you. To utilize the full power of multithreading, how threads communicate with each other is crucial. I will provide a more detailed description on this aspect.

**Comment by XavierQuan** [comment]
Changes have been made to include a more detailed instruction of how threads communication.

**Comment by Excelsior31** [comment]
The introduction to Multithreading is very detailed and is a helpful documentation to refer to when using multithreading. However, I think the majority of the documentation is conceptual, please consider adding some code examples in Java so that we can have a better understanding of how multithreading is implemented in code.

**Comment by ChenJiang03** [comment]
> > This is a very detailed explanation of multithreading. However, from my current understanding, data sharing between threads is a key point in correctly utilizing multithreading to its max potential. As you mentioned above, thread share file, heap, and code. Can you dive deeper into how threads share data between each other?
> 
> Thank you for your advice and I agree with you. To utilize the full power of multithreading, how threads communicate with each other is crucial. I will provide a more detailed description on this aspect.

Hi! No problem! Also, I forgot to mention there is a type in the word "thread" in your "thread class" section

File: Topics/Tech_Stacks/Multithreading.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	- Multithreaded if the tasks share a lot of data
+	- Threads are much faster to create an destroy
+	- Switching between threads of the same process is faster(shorter context switches)
+- Multi-Process
+	- Security and stability are of higher importance
+	- Tasks are unrelated to each other
+
+---
+
+## Tread Class
```
Comment (ChenJiang03) [inline Topics/Tech_Stacks/Multithreading.md:55]: Hi! Here is a typo! Please change it , thanks

Comment (XavierQuan) [reply Topics/Tech_Stacks/Multithreading.md:55]: Thank you for spotting it!

Comment (Excelsior31) [inline Topics/Tech_Stacks/Multithreading.md]: I find the concept of the Thread Class a bit unclear. Could you kindly provide some code examples to illustrate how to use it?

**Comment by XavierQuan** [comment]
> The introduction to Multithreading is very detailed and is a helpful documentation to refer to when using multithreading. However, I think the majority of the documentation is conceptual, please consider adding some code examples in Java so that we can have a better understanding of how multithreading is implemented in code.

Thank you for the suggestion, I will add some coding example and usage to my documentation.

**Review by Excelsior31 — CHANGES_REQUESTED** [review]

File: Topics/Tech_Stacks/Multithreading.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	- Multithreaded if the tasks share a lot of data
+	- Threads are much faster to create an destroy
+	- Switching between threads of the same process is faster(shorter context switches)
+- Multi-Process
+	- Security and stability are of higher importance
+	- Tasks are unrelated to each other
+
+---
+
+## Thread Class
```
[the remark this answers is not in the document]
Comment (XavierQuan) [reply Topics/Tech_Stacks/Multithreading.md:55]: Sure, I will add some code example showing how you would implement multithreading in a development senario

**Comment by ChenJiang03** [comment]
I think this is already a very helpful documentation. I have used multithreading in the past and I encountered some problems. For example, it is very common that we see race condition when multiple threads are accessing a shared resource. Consider adding some content about some common issues that programmers might face using multithreading, and its solution.

File: Topics/Tech_Stacks/Multithreading.md (hunk 1 above)
Comment (XavierQuan) [reply Topics/Tech_Stacks/Multithreading.md]: Corrected

**Review by Excelsior31 — APPROVED** [review]
Thank you for updating the documentation with practical code examples, it significantly enhances my understanding of multithreading in Java. Your detailed introduction to Multithreading is incredibly helpful, providing a solid foundation for understanding this complex topic. Great job on that!

**Comment by XavierQuan** [comment]
> I think this is already a very helpful documentation. I have used multithreading in the past and I encountered some problems. For example, it is very common that we see race condition when multiple threads are accessing a shared resource. Consider adding some content about some common issues that programmers might face using multithreading, and its solution.

You are correct, racing condition and deadlock are a common issues people might face during development. I will do some research on this part and update my documentation.

**Comment by XavierQuan** [comment]
Changes have been updated on the topic of common issues.

**Review by Excelsior31 — APPROVED** [review]
The final version of the Multithreading documentation looks fantastic and is extremely informative, offering a thorough and practical understanding of the topic. Great work!

**Review by ChenJiang03 — COMMENTED** [review]
It looks great! Good job

**Pull request merged**